inductiva/resources/machine_cluster.py

```diff
@@ -73,12 +74,26 @@ def __str__(self):
         return f"MPI Cluster {self.name} with {self.machine_type} " \
                f"x{self.num_machines} machines"
 
-    def start(self):
-        """Start the MPI Cluster."""
+    def start(self,
+              max_idle_time: float = None,
+              auto_terminate: Union[str, float] = None):
```

Comment:
A bit of inconsistency here: `max_idle_time` is in minutes but `auto_terminate` can be in hours.
In either case, I don't like any of them because they're hard to understand and give no indication of the units.

I suggest using existing python primitives :

```suggestion
              max_idle_time: timedelta = None,
              auto_terminate: datetime = None):
```